java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -242,116 +301,164 @@ JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getOutputInfo
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
-    OrtSession* session = (OrtSession*) sessionHandle;
-    OrtRunOptions* runOptions = (OrtRunOptions*) runOptionsHandle;
-
-    // Create the buffers for the Java input and output strings
-    const char** inputNames;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(char*)*numInputs,(void**)&inputNames));
-    const char** outputNames;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(char*)*numOutputs,(void**)&outputNames));
-    jobject* javaInputStrings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(jobject)*numInputs,(void**)&javaInputStrings));
-    jobject* javaOutputStrings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(jobject)*numOutputs,(void**)&javaOutputStrings));
-
-    // Extract a C array of longs which are pointers to the input tensors.
-    // Need to convert longs to OrtValue* in case we run on non-64bit systems
-    jlong* inputTensors = (*jniEnv)->GetLongArrayElements(jniEnv,tensorArr,NULL);
-    const OrtValue** inputValues;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(OrtValue*)*numInputs,(void**)&inputValues));
-
-    // Extract the names and native pointers of the input values.
-    for (int i = 0; i < numInputs; i++) {
-        javaInputStrings[i] = (*jniEnv)->GetObjectArrayElement(jniEnv,inputNamesArr,i);
-        inputNames[i] = (*jniEnv)->GetStringUTFChars(jniEnv,javaInputStrings[i],NULL);
-        inputValues[i] = (OrtValue*)inputTensors[i];
-    }
+JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_run(JNIEnv* jniEnv, jobject jobj, jlong apiHandle,
+                                                                  jlong sessionHandle, jlong allocatorHandle,
+                                                                  jobjectArray inputNamesArr, jlongArray tensorArr,
+                                                                  jlong numInputs, jobjectArray outputNamesArr,
+                                                                  jlong numOutputs, jlong runOptionsHandle) {
+
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+  OrtRunOptions* runOptions = (OrtRunOptions*)runOptionsHandle;
+
+  jobjectArray outputArray = NULL;
+
+  // Create the buffers for the Java input & output strings, and the input pointers
+  const char** inputNames = NULL;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char*) * numInputs, (void**)&inputNames));
+  if (code != ORT_OK) {
+    // Nothing to cleanup, return and throw exception
+    return outputArray;
+  }
```

Comment:
These and below are not required to be allocated with ORT allocator. It is much more easier to use malloc/free/

Reply:
Ok.